hasql/asyncsqlalchemy.py

```diff
@@ -70,4 +71,55 @@ def _driver_metrics(self) -> Sequence[DriverMetrics]:
         ]
 
 
-__all__ = ("PoolManager",)
+def async_sessionmaker(
+    pool_manager: PoolManager,
+    *,
+    class_: Type[AsyncSession] = AsyncSession,
+    autoflush: bool = True,
+    expire_on_commit: bool = True,
+    info: Optional[Dict[Any, Any]] = None,
+    acquire_kwargs: Optional[Dict[str, Any]] = None,
+    **kw: Any,
+) -> Callable[..., _AsyncGeneratorContextManager]:
+    """Create async session maker with hasql pool support.
+
+    This function replaces the default `async_sessionmaker` from
+    SQLAlchemy to work with the `PoolManager` class. It allows you to
+    create an async session that is bound to a connection acquired from
+    the pool. The session will automatically release the connection
+    back to the pool when the session is closed.
+
+    You also can specify the session class to use with the `class_`
+    parameter, and you can customize the session's behavior with
+    parameters like `autoflush`, `expire_on_commit`, and `info`.
+
+    Use the `acquire_kwargs` to pass additional parameters to the
+    `pool.acquire()` method. E.g. to create a session with replica
+    connection:
+    >>> ReplicaSession = async_sessionmaker(
+    >>>     pool_manager,
+    >>>     acquire_kwargs={"read_only": True}
+    >>> )
+
+    """
+    if acquire_kwargs is None:
+        acquire_kwargs = {}
+
+    @asynccontextmanager
+    async def create_async_session() -> AsyncIterator[AsyncSession]:
+        """Create an async session with connection from the pool."""
+        # TODO(PY310): Use parentheses to break the statement in multiple lines
+        async with pool_manager.acquire(**acquire_kwargs) as connection, \
```

Comment:
```suggestion
        async with (
            pool_manager.acquire(**acquire_kwargs) as connection,
            # more pythonic call it klass to avoid using a keyword class
            klass(
                bind=connection,
                autoflush=autoflush,
                expire_on_commit=expire_on_commit,
                info=info,
                **kw,
            ) as session
        ):
            yield session
```

Reply:
@mosquito, parentheses syntax available since py3.10 only :(

Reply:
I used `class_` name to save naming compatibility with SQLAlchemy's sessionmaker – it named `class_` there

Reply:
Probably the right time to drop 3.9 support ;-)